File: BosonClient.R
# BosonClient.R
# run: Rscript BosonClient.R
# run: aws batch submit-job --job-name boson-job --job-queue boson-job-queue --job-definition boson-batch-job --container-overrides '{"command":["sh","driver.sh","us-west-2","--bootstrap-jobs","5","s3://boson-base/rboson-base/boson_rjob_0.rdata","boson_rjob_0.rdata","s3://boson-base/tmp/"]}'

# load libraries & scripts
source('s3utils.R')
source('AWSBatchUtils.R')

# params
batch.id = 1
njobs = 4
s3.path = 's3://boson-base/rboson-test/'

#BosonTask = function (x, q) {
# return ( quantile (x, probs = q) )
#}

BosonTask = log
BosonParams = as.list(1:1000)
#for (i in 1:101) { BosonParams[[i]] = list(q = (i-1) / 100) }
#x = rnorm (10000, 0, 1)
#x = log(1:1000)

# upload metadata
SaveObjectesInS3(
  FUN = BosonTask, X = BosonParams, extra.args = list(),
  s3.path = s3.path,
  key = paste0('batch_', batch.id, '_in')
)

# submit a batch job
SubmitBatchJobs(batch.id = batch.id, njobs = njobs, s3.path = s3.path)

# download outcomes
S3ListFiles = function (s3path = s3.path)

out.all = LoadObjectsFromS3(
  s3.path = s3.path,
  keys = paste0('batch_', batch.id, '_out_', 1:njobs)
)

File: BosonClient.py
# BosonClient.py
# run: python BosonClient.py
# run: aws batch submit-job --job-name boson-job --job-queue boson-job-queue --job-definition boson-batch-job --container-overrides '{"command":["sh","driver.sh","us-west-2","--bootstrap-jobs","5","s3://boson-base/rboson-base/boson_rjob_0.rdata","boson_rjob_0.rdata","s3://boson-base/tmp/"]}'

# load libraries & scripts
from s3utils import * 
from AWSBatchUtils import *

# params
batch_id = 1
njobs = 4
s3_path = 's3://boson-base/pyboson-test/'

def BosonTask(w, x, y, z):
	import math
	return(w * math.log(x) + y - z)
# BosonTask(x = 1, y = 2, z = 3)

BosonParams = []; y = 10; z = 10; w = 1
for i in range(100):
	BosonParams.append({'x':i, 'y':y})

# upload metadata
SaveObjectesInS3(s3_path,
	key = 'batch_{}_in'.format(batch_id),
	FUN = BosonTask, X = BosonParams, w = w, z = z)
import sys
sys.exit(0)

# submit a batch job
SubmitBatchJobs(batch_id = batch_id, njobs = njobs, s3_path = s3_path)

# download outcomes
S3ListFiles(s3_path)

out_all = LoadObjectsFromS3( s3_path = s3_path, keys = ['batch_{}_out_'.format(batch_id) + s for s in map(str,range(1,njobs))] )


